pandas/tests/test_strings.py

```diff
@@ -2760,6 +2761,26 @@ def test_str_cat_raises_intuitive_error(self):
         with tm.assert_raises_regex(ValueError, message):
             s.str.cat('    ')
 
+    def test_str_cat_align(self):
+        # https://github.com/pandas-dev/pandas/issues/18657
+        base = Series(['a', 'b', 'c', 'd', 'e'])
+        r = base.reindex([0, 1, 2, 3])
+        s = base.reindex([1, 3, 0, 2])
+        t = base.reindex([3, 0, 4, 1])
+        expect_rs_aligned = Series(['aa', 'bb', 'cc', 'dd'])
+        expect_rs_unaligned = Series(['ab', 'bd', 'ca', 'dc'])
```

Comment:
put a blank line between cases. Add a comment to cases if needed.  If you are writing things than once, pls parameterize.

Reply:
Overlooked this, will change in next commit

Reply:
I updated the tests, so that they are separated by lines, commented where necessary, and much easier to read.